act/utils/data_utils.py

```diff
@@ -53,3 +53,68 @@ def add_in_nan(time, data):
     d_data = xr.DataArray(d_data)
 
     return d_time, d_data
+
+
+def get_missing_value(data_object, variable, default=-9999, add_if_missing_in_obj=False,
+        use_FillValue=False):
+    '''Function to get missing value from missing_value or _FillValue attribute.
+    Works well with catching errors and allows for a default value when a missing
+    value is not listed in the object.
+
+    Parameters
+    ----------
+    data_object : obj
+        Data object to search.
+    variable : str
+        Variable name to use for getting missing value. 
+    default : int or float or None
+        Default value to use if missing value attribute is not in data_object.
+        If no value should be retuned if missing_value or _FillValue is not 
+        set as attribute in object, set default=None.
+    add_if_missing_in_obj : bool
+        Boolean to add to object if does not exist.
+    use_FillValue : bool
+        Boolean to use _FillValue instead of missing_value. If missing_value
+        does exist and _FillValue does not with add_if_missing_in_obj set to 
+        True, will add _FillValue set to missing_value value.
+
+    Returns
+    -------
+    missing : scalar int or float
+        Value used to indicate missing value matching type of data.
+
+    Examples
+    --------
+    >>> missing = get_missing_value(dq_object, 'temp_mean')
+    >>> missing
+    -9999.0
+
+    '''
+    in_object = False
+    if use_FillValue:
+        missing_atts = ['_FillValue','missing_value']
+    else:
+        missing_atts = ['missing_value','_FillValue']
+
+    for att in missing_atts:
+        try:
+            missing = data_object[variable].attrs[att]
+            in_object = True
+            break
+        except (AttributeError, KeyError):
+            missing = default
+
+    # Check data type and try to match missing_value to the data type of data
+    try:
+        missing = data_object[variable].data.dtype.type(missing)
+    except Exception as error:
```

Comment:
It is generally not a good idea to do try/except statements unless you are catching specific errors instead of just skipping code for any error. I think it would be better to have the routine add the missing value attribute before this line with the name specified by the user if one is not found.